x-pack/plugins/reporting/server/lib/store/store.ts

```diff
@@ -309,22 +321,27 @@ export class ReportingStore {
       status: statuses.JOB_STATUS_FAILED,
     });
 
+    let body: UpdateResponse<ReportDocument>;
     try {
       const client = await this.getClient();
-      const { body } = await client.update<ReportDocument>({
-        id: report._id,
-        index: report._index,
-        if_seq_no: report._seq_no,
-        if_primary_term: report._primary_term,
-        refresh: true,
-        body: { doc },
-      });
-      return body;
+      body = (
+        await client.update<ReportDocument>({
+          id: report._id,
+          index: report._index,
+          if_seq_no: report._seq_no,
+          if_primary_term: report._primary_term,
+          refresh: true,
+          body: { doc },
+        })
+      ).body;
     } catch (err) {
       this.logger.error(`Error in updating status to failed! Report: ` + jobDebugMessage(report));
       this.logger.error(err);
       throw err;
     }
+
+    getEventLogger(this.reportingCore, report).logReportFailure(`Report ${report._id} failed.`);
```

Comment:
What do you think about sending more information about the report object in a structured way, e.g.:

```suggestion
    getEventLogger(this.reportingCore, report).logReportFailure();
```

and internally it would:

```ts
    logReportFailure(message: string): FailedReport {
      const event = deepMerge(
        {
          message: "Report failed.", // Or maybe we skip this field altogether!
          event: { kind: 'event', action: ActionType.FAIL_REPORT },
          log: { level: 'info' },
        } as Partial<FailedReport>,
        this.eventObj
      );

      genericLogger.logEvent(event);
      return event;
    }
```

the logger should already know to which report it is "associated", because of the `report` param in `getEventLogger`:

```ts
getEventLogger(this.reportingCore, report)
```

Reply:
https://github.com/elastic/kibana/pull/119923/commits/59213c150cdbc34ffb27c2858946215acbfa6593